### PR TITLE
Add command to determine Reddit sentiment about a ticker/company.

### PR DESCRIPTION
**Pull request by jbushago**
# Description

This PR adds a new command `reddit_sent` to the terminal under `stocks/ba`. This command searches through subreddits for a given ticker (or company name), aggregates/preprocesses its findings, and then determines the overall sentiment about the query (with an optional graphic).

It's a quick way to get a general understanding about how different communities feel about a company/ticker. An example usage is shown below:
image

No new dependencies are needed for the command.

# How has this been tested?

All tests pass after these changes.

# Others

@cweckerl, @montoys, and I all worked on the implementation and review.

### CODE REVIEW

File: openbb_terminal/common/behavioural_analysis/reddit_view.py [hunk 1]
```diff
@@ -323,3 +326,81 @@ def display_due_diligence(
...
+        sentiment = reddit_model.get_sentiment(preprocessed_text)
+        polarity_scores.append(sentiment)
+        post_data.append([p.title, sentiment])
+
+    avg_polarity = sum(polarity_scores) / len(polarity_scores)
+
+    columns = ["Title", "Polarity Score"]
+    df = pd.DataFrame(post_data, columns=columns)
+
+    if graphic:
```
Comment (piiq) [inline openbb_terminal/common/behavioural_analysis/reddit_view.py:394]: [critical] this is the only critical comment for the PR.
the way the plot is created and used here leads to axes overlapping each other on the same figure if you call the command multiple times consecutively. 
image

in addition there's no support for external axes which makes it impossible to use the chart in notebooks and reports.

to resolve this you'll need to create a use seaborn itself in external axis mode. there are a couple of examples in the code of how the chart that uses seaborn can be implemented. imo this is the simplest one:
https://github.com/OpenBB-finance/OpenBBTerminal/blob/71cd1023d4d41afff50d89c6b7869027837564b9/openbb_terminal/stocks/comparison_analysis/yahoo_finance_view.py#L203-L228

note the `ax=ax` kwarg going to the sns call and use of the `theme` helper

**Review by piiq — CHANGES_REQUESTED** [review]
@jbushago @cweckerl @montoys you are awesome! 🚀🚀🚀
Thank you for this contribution. I've already started playing with the functions that you've added (and thinking about incorporating them to my routines ... hehe)

I have left a bunch of suggestions in the code. Not mandatory

But there's one thing related to the sentiment chart that I would like you to change (see comment). Feel free to tag me if you have any questions

File: openbb_terminal/common/behavioural_analysis/reddit_model.py [hunk 2]
```diff
@@ -19,6 +22,8 @@
 
 logger = logging.getLogger(__name__)
 
+analyzer = SentimentIntensityAnalyzer()
```
[the remark this answers is not in the document]
Comment (jbushago) [reply openbb_terminal/common/behavioural_analysis/reddit_model.py]: Resolved in [d4f347f](https://github.com/OpenBB-finance/OpenBBTerminal/pull/1744/commits/d4f347ff2a89a62498bd865af53e98c1e4168443).

File: openbb_terminal/common/behavioural_analysis/reddit_model.py [hunk 3]
```diff
@@ -567,3 +572,153 @@ def get_due_dilligence(
             console.print(f"[red]Invalid response: {str(e)}[/red]\n")
 
     return subs
+
+
+@log_start_end(log=logger)
+def get_posts_about(
+    ticker: str,
+    limit: int,
+    sort: str,
```
[the remark this answers is not in the document]
Comment (jbushago) [reply openbb_terminal/common/behavioural_analysis/reddit_model.py]: Resolved in [e39efbe](https://github.com/OpenBB-finance/OpenBBTerminal/pull/1744/commits/e39efbe3b4afeb38f542564b6129647283b6a384).

File: openbb_terminal/common/behavioural_analysis/reddit_model.py [hunk 4]
```diff
@@ -567,3 +572,153 @@ def get_due_dilligence(
...
+
+    Returns
+    -------
+    float
+        A number in the range [-1, 1] representing sentiment
+    """
+    post_data_l = " ".join(post_data)
+    sentiment = analyzer.polarity_scores(post_data_l)
+    pre_norm = sentiment["pos"] - sentiment["neg"]
+    norm = (pre_norm - 0.06) * 8  # Explicitly give us a better range
```
[the remark this answers is not in the document]
Comment (jbushago) [reply openbb_terminal/common/behavioural_analysis/reddit_model.py]: These constants are empirical, I added an extended comment about that in the function :).

File: openbb_terminal/common/behavioural_analysis/reddit_view.py [hunk 5]
```diff
@@ -323,3 +326,81 @@ def display_due_diligence(
...
+            texts.extend(tlcs)
+        preprocessed_text = reddit_model.clean_reddit_text(texts)
+        sentiment = reddit_model.get_sentiment(preprocessed_text)
+        polarity_scores.append(sentiment)
+        post_data.append([p.title, sentiment])
+
+    avg_polarity = sum(polarity_scores) / len(polarity_scores)
+
+    columns = ["Title", "Polarity Score"]
+    df = pd.DataFrame(post_data, columns=columns)
```
[the remark this answers is not in the document]
Comment (jbushago) [reply openbb_terminal/common/behavioural_analysis/reddit_view.py:401]: Added an option for that in [f2558a8](https://github.com/OpenBB-finance/OpenBBTerminal/pull/1744/commits/f2558a813cf922c8d07f8ca169bd5b26b46c245f).

File: openbb_terminal/common/behavioural_analysis/reddit_view.py (hunk 1 above)
Comment (jbushago) [reply openbb_terminal/common/behavioural_analysis/reddit_view.py:406]: Great catch here! Hopefully this is fixed by [ac8d2c6](https://github.com/OpenBB-finance/OpenBBTerminal/pull/1744/commits/ac8d2c60aaf03f4b63a1082116bf21756b172dd9), but let me know if I did anything wrong.

**Pull request merged**